sale_order_line_multi_warehouse/models/sale_order.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2024 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    allow_sale_multi_warehouse = fields.Boolean(
+        compute="_compute_allow_sale_multi_warehouse", store=True
+    )
+    suitable_warehouse_ids = fields.Many2many(
+        string="Suitable Warehouses",
+        comodel_name="stock.warehouse",
+        compute="_compute_suitable_warehouse_ids",
+    )
+
+    @api.depends("company_id")
+    def _compute_allow_sale_multi_warehouse(self):
+        for order in self:
+            order.allow_sale_multi_warehouse = (
+                order.company_id.allow_sale_multi_warehouse
+            )
+
+    @api.depends("warehouse_id")
+    def _compute_suitable_warehouse_ids(self):
+        for order in self:
+            order.suitable_warehouse_ids = (
+                order.warehouse_id + order.warehouse_id.alternative_warehouse_ids
+            )
+
+    def get_incompatible_multi_warehouse_lines(self, warehouse):
+        self.ensure_one()
+        compatible_warehouse_ids = warehouse + warehouse.alternative_warehouse_ids
+        return self.order_line.mapped("sale_order_line_warehouse_ids").filtered(
+            lambda a: a.warehouse_id.id not in compatible_warehouse_ids.ids
+        )
```

Comment:
I think you can use:

`return self.order_line.sale_order_line_warehouse_ids - self.suitable_warehouse_ids`